### PR TITLE
anonymous signIn done✓

### PR DESCRIPTION
**Pull request by Fausto95**
- [x] Already covered by tests
- [ ] Manual testing
- [x] Unit testing

### CODE REVIEW

File: src/redux/actions/authentication.js [hunk 1]
```diff
@@ -55,6 +56,18 @@ export const signIn = (token: string): StoreAction<Action> => async dispatch =>
   }
 };
 
+export const signInAnonymous = (): StoreAction<Action> => async dispatch => {
+  const {_bodyInit: token} = await fetch('https://up.coorpacademy.com/api/v1/anonymous/mobile', {
```
Comment (Fausto95) [inline src/redux/actions/authentication.js]: It's bizarre that I need to do this `const {_bodyInit: token} = ...` in order to get the response

File: src/screens/home.js [hunk 2]
```diff
@@ -63,7 +63,7 @@ class HomeScreen extends React.PureComponent<Props, State> {
     const action = this.props.onLogoLongPress && (await this.props.onLogoLongPress());
     // When we log out, we reset the entirely stack
     // To avoid router directions and other stuffs
-    if (action) this.props.navigation.reset([NavigationActions.navigate({routeName: 'Splash'})], 0);
```
Comment (Fausto95) [inline src/screens/home.js]: By the way @VincentCATILLON, IDK why you've refactored this code, but this is not the good way to handle this!
Whenever something goes wrong we're going to stuck in the splash's screen blue dead color

Comment (Fausto95) [reply src/screens/home.js]: When logging out, we do delete every user's local data, so we are sure that we do not have any stored token, redirecting the user to the splash screen is useless

File: src/components/authentication.js [hunk 3]
```diff
@@ -108,11 +114,22 @@ const Authentication = ({onPress, onAssistancePress}: Props) => (
...
-        <Space type={SPACE.SMALL} />
+        <Space type={SPACE.BASE} />
+        <View style={styles.centeredContent}>
+          <Html
+            fontSize={theme.fontSize.large}
+            style={styles.startDemo}
+            onLinkPress={onStartDemoPress}
+            anchorTextColor={theme.colors.white}
+          >
+            {`No learning platform? <a href="/"><b><u>Start demo</u></b></a>`}
```
Comment (Fausto95) [inline src/components/authentication.js]: Must pull translation from player web

Comment (VincentCATILLON) [reply src/components/authentication.js]: Create a translation key for this

File: src/components/authentication.js [hunk 4]
```diff
@@ -75,6 +76,11 @@ const styles = StyleSheet.create({
     fontSize: theme.fontSize.medium,
     color: theme.colors.white
   },
+  startDemo: {
+    margin: 0,
```
Comment (VincentCATILLON) [inline src/components/authentication.js]: Is this really useful ?

File: src/redux/actions/authentication.js [hunk 5]
```diff
@@ -55,6 +56,15 @@ export const signIn = (token: string): StoreAction<Action> => async dispatch =>
   }
 };
 
+export const signInAnonymous = (): StoreAction<Action> => async dispatch => {
+  const response = await fetch('https://up.coorpacademy.com/api/v1/anonymous/mobile', {
+    method: 'POST'
+  });
+  const token = await response.text();
+  // $FlowFixMe
+  return signIn(token)(dispatch);
```
Comment (VincentCATILLON) [inline src/redux/actions/authentication.js]: should not be `return dispatch(signIn(token))` ?

File: src/screens/authentication.js [hunk 6]
```diff
@@ -43,14 +53,19 @@ class AuthenticationScreen extends React.PureComponent<Props> {
     return (
       <Screen testID="authentication-screen" noScroll noSafeArea>
         <StatusBar barStyle="light-content" backgroundColor={TOP_COLOR} />
-        <Authentication onPress={this.handlePress} onAssistancePress={this.handleAssistancePress} />
+        <Authentication
+          onStartDemoPress={this.handleSignInAnonymous}
```
Comment (VincentCATILLON) [inline src/screens/authentication.js]: handleStartDemoPress

File: src/screens/home.js [hunk 7]
```diff
@@ -63,7 +63,8 @@ class HomeScreen extends React.PureComponent<Props, State> {
...
+    if (action)
+      this.props.navigation.reset([NavigationActions.navigate({routeName: 'Authentication'})], 0);
```
Comment (VincentCATILLON) [inline src/screens/home.js]: https://github.com/CoorpAcademy/mobile/pull/146/commits/7d0fda7720c8a7a3c29dd673a7f3c595b9f5cbf2#diff-a77f71d8a99d3eec962380c09d1ae727R53 ;)

Comment (Fausto95) [reply src/screens/home.js]: Yea, @VincentCATILLON , my bad, squashing commits makes it only appear the pr author as the main man

File: src/redux/actions/authentication.js (hunk 5 above)
Comment (Fausto95) [reply src/redux/actions/authentication.js]: That's a great shortcut, but dont work for tests!

**Pull request merged**